Introduce MAX32650 SoC

yasinustunerg opened the pull request:
This PR:

Adds [MAX32650 ](https://www.analog.com/en/products/max32650.html) SoC basic components
Adds [MAX32650EVKIT](https://www.analog.com/en/resources/evaluation-hardware-and-software/evaluation-boards-kits/max32650-evkit.html) basic components.
Adds [MAX32650FTHR](https://www.analog.com/en/resources/evaluation-hardware-and-software/evaluation-boards-kits/max32650fthr.html) basic components.
Adds [AD_SWIOT1L_SL](https://www.analog.com/en/resources/evaluation-hardware-and-software/evaluation-boards-kits/AD-SWIOT1L-SL.html) basic components.

Signed-off-by: Yasin Ustuner <Yasin.Ustuner@analog.com>
Signed-off-by: Furkan Akkiz <hasanfurkan.akkiz@analog.com>
Signed-off-by: Burak Babaoglu <burak.babaoglu@analog.com>

Burakbabao committed:
soc: adi: Add the MAX32650 SoC

This commit adds MAX32650 Kconfig
and dts files for basic port.

Signed-off-by: Burak Babaoglu <Burak.Babaoglu@analog.com>
Signed-off-by: Yasin Ustuner <Yasin.Ustuner@analog.com>

Burakbabao committed:
boards: adi: Add MAX32650EVKIT board

This commit adds MAX32650EVKIT board
basic port.

Signed-off-by: Burak Babaoglu <Burak.Babaoglu@analog.com>
Signed-off-by: Yasin Ustuner <Yasin.Ustuner@analog.com>

Commit:
drivers: serial: Change Tx Status macro for MAX32

TX status macro is different for MAX32650 SoC.
The common macro is added to the hal.
This commit changes the tx status macro with
the common one.

Signed-off-by: Yasin Ustuner <Yasin.Ustuner@analog.com>

Commit:
manifest: Add MAX32650 hal files

This commit adds necessary hal files for
MAX32650 board.

Signed-off-by: Yasin Ustuner <Yasin.Ustuner@analog.com>

jilaypandya commented on boards/adi/max32650evkit/doc/index.rst:
```diff
@@ -0,0 +1,68 @@
+
+Overview
+********
+The MAX32650 evalauation kit (EV kit) provides a platform for evaluating the
```
```suggestion
The MAX32650 evaluation kit (EV kit) provides a platform for evaluating the
```

yasinustunerg replied:
Done

MaureenHelm commented on west.yml:
```diff
@@ -142,7 +142,7 @@ manifest:
         - fs
     - name: hal_adi
-      revision: a9ca7d172152789a02439bbdfb94feb72e30553d
+      revision: pull/16/head
```
zephyrproject-rtos/hal_adi#16 has been merged, so this can be updated to the new SHA

MaureenHelm commented on drivers/serial/uart_max32.c:
```diff
@@ -376,7 +376,7 @@ static int api_irq_tx_ready(const struct device *dev)
 
 	return ((inten & (ADI_MAX32_UART_INT_TX | ADI_MAX32_UART_INT_TX_OEM)) &&
-		!(data->status & MXC_F_UART_STATUS_TX_FULL));
+		!(data->status & ADI_MAX32_UART_STATUS_TX_FULL));
```
```
drivers: serial: Change Tx Status macro

TX status macro is different for MAX32650 SoC.
The common macro is added to the hal.
This commit changes the tx status macro with
the common one.

Signed-off-by: Yasin Ustuner <Yasin.Ustuner@analog.com>
```
It would be helpful for the subject line to mention max32

yasinustunerg replied:
Done

hfakkiz committed:
boards: adi: Add AD_SWIOT1L_SL board

This commit adds AD_SWIOT1L_SL board basic port.

Signed-off-by: Furkan Akkiz <hasanfurkan.akkiz@analog.com>

yasinustunerg commented on boards/adi/max32650evkit/doc/index.rst:
```diff
@@ -0,0 +1,110 @@
+Supported Features
+==================
+
+The ``max32650evkit`` board target supports the following interfaces:
```
Done

ozersa commented on boards/adi/max32650evkit/max32650evkit.dts:
```diff
@@ -0,0 +1,93 @@
+	status = "okay";
+};
+
+&clk_iso {
+	status = "okay";
+};
```
Why the ISO clock need to be enabled on default?

yasinustunerg replied:
Removed

yasinustunerg commented on dts/arm/adi/max32/max32650.dtsi:
```diff
@@ -0,0 +1,109 @@
+};
+
+&clk_iso {
+	clock-frequency = <DT_FREQ_M(50)>;
```
For the latest revisions, 50 MHz is used.
![image](https://github.com/user-attachments/assets/1bacc28c-6dbe-4d96-9abd-e81222509cb1)
![image](https://github.com/user-attachments/assets/51c42ef4-9ca0-4a2e-845c-52b50ded46af)

ozersa commented on boards/adi/max32650evkit/doc/index.rst:
```diff
@@ -0,0 +1,119 @@
+
+    - 120MHz Arm Cortex-M4 with FPU
+    - SmartDMA Provides Background Memory Transfers with Programmable Data Processing
+    - 120MHz High-Speed and 40MHz Low-Power Oscillators
```
50MHz ?
https://github.com/zephyrproject-rtos/zephyr/pull/85572#discussion_r1952617586

yasinustunerg replied:
Done

ozersa commented on boards/adi/max32650fthr/doc/index.rst:
```diff
@@ -0,0 +1,118 @@
+
+    - 120MHz Arm Cortex-M4 with FPU
+    - SmartDMA Provides Background Memory Transfers with Programmable Data Processing
+    - 120MHz High-Speed and 40MHz Low-Power Oscillators
```
50MHz ?
https://github.com/zephyrproject-rtos/zephyr/pull/85572#discussion_r1952617586

Commit:
boards: adi: Add MAX32650FTHR board

This commit adds MAX32650FTHR board
basic port.

Signed-off-by: Yasin Ustuner <Yasin.Ustuner@analog.com>
Signed-off-by: Burak Babaoglu <Burak.Babaoglu@analog.com>

hfakkiz committed:
drivers: hwinfo: Update driver for MAX32650

There are function mismatches for MAX32650. Because of that, created a
wrap version of the function to handle these differences.

Signed-off-by: Furkan Akkiz <hasanfurkan.akkiz@analog.com>

ttmut commented on boards/adi/ad_swiot1l_sl/ad_swiot1l_sl.yaml:
```diff
@@ -0,0 +1,14 @@
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
```
```suggestion
```
xtools is deprecated.

ttmut commented on boards/adi/max32650evkit/max32650evkit.yaml:
```diff
@@ -0,0 +1,14 @@
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
```
```suggestion
```

yasinustunerg replied:
Done

ttmut commented on boards/adi/max32650fthr/max32650fthr.yaml:
```diff
@@ -0,0 +1,14 @@
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
```
```suggestion
```

ttmut reviewed (changes requested):
xtools is deprecated. See https://github.com/zephyrproject-rtos/zephyr/pull/80466 and https://github.com/zephyrproject-rtos/zephyr/pull/85725.

MaureenHelm commented:
@ttmut please revisit
